controllers/drpolicy_controller.go

```diff
@@ -108,18 +141,142 @@ func (r *DRPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
-	// Check if the MirrorPeer contains StorageClient reference
-	hasStorageClientRef, err := utils.IsStorageClientType(ctx, r.HubClient, *mirrorPeer, false)
-	if err != nil {
-		logger.Error("Failed to determine if MirrorPeer contains StorageClient reference", "error", err)
-		return ctrl.Result{}, err
+	if mirrorPeer.Spec.Type != multiclusterv1alpha1.Async {
+		return ctrl.Result{}, nil
 	}
 
-	if hasStorageClientRef {
-		logger.Info("MirrorPeer contains StorageClient reference. Skipping creation of VolumeReplicationClasses", "MirrorPeer", mirrorPeer.Name)
-		return ctrl.Result{}, nil
+	err = r.createOrUpdateManifestWorkForVRC(ctx, mirrorPeer, &drpolicy)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create VolumeReplicationClass via ManifestWork: %v", err)
```

Comment:
```suggestion
		return ctrl.Result{}, fmt.Errorf("failed to create VolumeReplicationClass template via ManifestWork: %v", err)
```

nit

Reply:
It's a small detail. Can I fix it when I work on that file next time?

Reply:
Sure !